cmd: copy edits for import-url ref.

utkarshsingh99 opened the pull request:
- [x] ❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

- [x] 🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

- [x] Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Partially address #1469 
- [x] import, import-url,

jorgeorpinel commented on content/docs/command-reference/add.md:
```diff
@@ -158,8 +158,9 @@ outs:
 
-This is a standard `.dvc` file with only one output (`outs` field). The hash
-value (`md5` field) corresponds to a file path in the <abbr>cache</abbr>.
+This is a standard `.dvc` file with only one output (`outs` field). The `outs`
+field can have any hash type (`md5` field in this case). The hash value
+corresponds to a file path in the <abbr>cache</abbr>.
```
Hmm... No. In this case it's `md5` specifically. Look at the YAML example above.

jorgeorpinel commented on content/docs/command-reference/import-url.md:
```diff
@@ -186,7 +186,7 @@ Let's take a look at the changes to the `data.xml.dvc`:
 [ETag](https://en.wikipedia.org/wiki/HTTP_ETag) recorded from the HTTP request.
 If the remote file changes, its ETag will be different. This metadata allows DVC
-to determine whether its necessary to download it again.
+to determine whether it is necessary to download it again.
```
Thanks. Let's do
```suggestion
to determine whether it's necessary to download it again.
```

jorgeorpinel commented:
You could add the "This" -> "These" update here 😉

utkarshsingh99 commented:
Fix #1492

utkarshsingh99 commented:
@jorgeorpinel I found `import-url.md` to be the best place for now to differentiate file hashes since all different remote locations are talked about in detail here including their requirements (such as strong ETag, etc) in DVC.
Since `import.md` (`import` command) just focuses on the DVC repos or Git repos, they will only have an md5 field, so I guess it isn't worth talking about different file hashes there.
What do you think?

utkarshsingh99 commented on content/docs/command-reference/import-url.md:
```diff
@@ -51,7 +51,7 @@ corresponding local path in the <abbr>workspace</abbr>. It records enough
 the local copy is out of date.
 
-`dvc repro` doesn't check and/or update generated `.dvc` files, use `dvc update`
+`dvc repro` doesn't check and/or update generated `.dvc` files. Use `dvc update`
```
Just thought that the two would be more readable here given that their subjects are different.

jorgeorpinel replied:
No, these are separate sentences. Also, out of scope.

jorgeorpinel replied:
"and/or" should just be "or" though, please change that since we're looking at this already.

utkarshsingh99 commented on content/docs/command-reference/import-url.md:
```diff
@@ -109,8 +118,8 @@ $ dvc run -n download_data \
-[`.dvc` file](/doc/user-guide/dvc-files-and-directories#dvc-files) and `dvc run`
-a regular stage (in
+[`.dvc` file](/doc/user-guide/dvc-files-and-directories#dvc-files), and
+`dvc run` generates a regular stage (in
```
Readability improvement. I think it can be changed back to original too, was just a potential enhancement.
Or we could also add 
```
..., while `dvc run` generates...
```

jorgeorpinel replied:
Out of scope. Please roll back.

I would decide now since you already noticed this paragraph but we are going to get rid of the "import stage" term soon so this whole sentence will have to be rewritten anyway.

jorgeorpinel replied:
BTW, appreciate the initiative! It's just that right now we want to have very small and focused PRs during this part of the process 🙂

utkarshsingh99 commented on content/docs/command-reference/import-url.md:
```diff
@@ -186,14 +195,11 @@ Let's take a look at the changes to the `data.xml.dvc`:
 
-You may want to get out of and remove the `example-get-started/` directory after
-trying this example (especially if trying out the following one).
-
```
I believe this statement is irrelevant here since anyone could arrive at this page without going through the `User Guide/Getting-Started` and it does not hold a direct info about the `import-url` command. It seems like the content was borrowed from one of those pages.

jorgeorpinel replied:
I don't think you followed the entire examples section first. That would need to be done to determine this. Feel free to do so and submit a different PR if you find this unnecessary, but it's out of scope for this PR. Please roll back here. Thanks

utkarshsingh99 replied:
Alright.
But do you think it should be removed? 
> I don't think you followed the entire examples section first.  

Actually, since this is in `command reference`, I thought it would be good to keep this file independent of the `Getting Started` module

jorgeorpinel replied:
I don't think it should be removed. But like I mentioned, I would need to follow the entire examples section first to decide.

> would be good to keep this file independent of the Getting Started

It is. Again, I don't think you read the whole examples section 🙂 

Please roll back.

utkarshsingh99 replied:
Oh okay. Now I get it. It's just related to the example above.
My apologies :smile:

utkarshsingh99 commented:
@jorgeorpinel Is there anything more I should do in this?
Please let me know.

jorgeorpinel commented:
> Partially address #1469

Please say specifically which part of that issue is addressed in the description of this PR (you can just copy the entire import/import-url checkbox here, and mark it).

> Fix #1492

FYI special keywords only work on the PR description. Not needed here anyway because I already closed that issue as fixed.

> I found import-url.md to be the best place for now to differentiate file hashes since all different remote locations are talked about

I see that import only has `md5` in its examples, indeed. What about "add (external dep)" and "run -d (external dep)"? (The other checkboxes in #https://github.com/iterative/dvc.org/issues/1469) Did you check them? You don't have to for this PR, I'm just curious.

skshetry commented on content/docs/command-reference/import-url.md:
```diff
@@ -88,6 +88,15 @@ Specific explanations:
+The `deps` field in the generated `.dvc` file contains a hash value along with
+the `path` to track changes in the remote file/directory. The hash type depends
+on the type of remote location (protocol) being tracked. For instance,
+files/directories tracked from HTTP(s)/S3/Azure/GS remote locations will have an
```
I made a mistake with the comment. GS uses `md5`. Sorry about that.

utkarshsingh99 replied:
Okay, I'll change it in the DVC Files and Directories page in the next commit :v:

jorgeorpinel replied:
No worries @skshetry, glad you caught it though! Thanks for the followup. I updated #1469 and fixed the dvc.lock doc in another PR. But this brings some questions... I'll ping you somewhere else.

@utkarshsingh99 please update the paragraph above accordingly...

jorgeorpinel commented on content/docs/command-reference/import-url.md:
```diff
@@ -88,6 +88,15 @@ Specific explanations:
   running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
   URL by appending `/path/to/file` to the `myremote`'s configured base path.
 
+The `deps` field in the generated `.dvc` file contains a hash value along with
+the `path` to track changes in the remote file/directory. The hash type depends
+on the type of remote location (protocol) being tracked. For instance,
+files/directories tracked from HTTP(s)/S3/Azure/GS remote locations will have an
+[ETag](https://en.wikipedia.org/wiki/HTTP_ETag) value in the `deps` field, while
+files/directories tracked from SSH/local locations will have an `md5` hash
+value. Files from HDFS remote locations use a `checksum` key in the `deps`
+field.
+
```
Ah thanks! But the intention of the ticket wasn't to repeat all this info here! Just double check that any existing mentions of hash values are correct. We should link "`.dvc` files" to the corresponding user guide though, so that people can see the full spec there, including types of hash values.

jorgeorpinel replied:
Also in line 111 can you capitalize the first word? "It". Thanks

utkarshsingh99 replied:
I see this link containing the info we need but how can I link precisely that paragraph which talks about the different hash values instead of the section `dvc.lock file`?
https://dvc.org/doc/user-guide/dvc-files-and-directories#dvclock-file

It would be great if we could link this paragraph:
> Regular dependencies and all kinds of outputs (including metrics and plots files) are also listed (per stage) in dvc.lock, but with an additional field to store the hash value of each file or directory tracked by DVC: md5 for local file system dependencies and SSH external dependencies, etag for HTTP, S3, Azure, and Google Cloud, and checksum for HDFS.

jorgeorpinel replied:
Can't link to a paragraph haha. That's not necessary, the anchor is enough... People have eyes 😉

jorgeorpinel replied:
This p is still there... Updated https://github.com/iterative/dvc.org/pull/1494#pullrequestreview-442823296. Resolving here.

jorgeorpinel commented:
> anything more I should do in this?

Yes, can you please change this text to use semicolons `;` in https://dvc.org/doc/user-guide/dvc-files-and-directories#dvclock-file ?
"directory tracked by DVC: md5 for local file system dependencies and SSH external dependencies; etag for HTTP, S3, Azure, and Google Cloud; and checksum for HDFS."

Also, in line 111 can you capitalize the first word? "It". Thanks for the other couple small typo fixes too, BTW!

utkarshsingh99 commented:
> What about "add (external dep)" and "run -d (external dep)"? (The other checkboxes in ##1469) Did you check them?

There's one mention of adding external files in `dvc add` but it's linking already to the External Dependencies page. Perhaps we can add a link to hash values' info here: [dvc add example](https://dvc.org/doc/command-reference/add#example-single-file)?

jorgeorpinel commented on content/docs/user-guide/dvc-files-and-directories.md:
```diff
@@ -211,9 +211,9 @@ Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
-directory tracked by DVC: `md5` for local file system dependencies and SSH
-<abbr>external dependencies</abbr>, `etag` for HTTP, S3, Azure, and Google
-Cloud, and `checksum` for HDFS.
+directory tracked by DVC: `md5` for local file system dependencies and SSH and
+Google Cloud <abbr>external dependencies</abbr>; `etag` for HTTP, S3, and Azure;
+and `checksum` for HDFS.
```
"and SSH and Google Cloud"? Too many ands.

Anyway, thanks but please roll this back, it's out of scope (I already did it elsewhere). Let's try to focus pls.

jorgeorpinel replied:
Ohhhh wait I see you started editing this file because I asked for the semicolons. Thanks! Yeah my bad, that was out of scope sorry. Please roll this back if you don't mind because I've already changed all this in #1527.

jorgeorpinel commented:
Nah actually add is a special case. Those aren't deps, they're outs. Don't worry about that for now.

But what about `dvc run -d`? I assume you didn't check it and that's fine. Let's just focus on addressing the remaining feedback up here ☝️ please, so we can merge this.

utkarshsingh99 commented:
Removed this
> For instance, files/directories tracked from HTTP(s)/S3/Azure remote locations will have an
[ETag](https://en.wikipedia.org/wiki/HTTP_ETag) value in the `deps` field, while
files/directories tracked from GS/SSH/local locations will have an `md5` hash
value. Files from HDFS remote locations use a `checksum` key in the `deps`
field.

Added:
> See [`dvc.lock` file](/doc/user-guide/dvc-files-and-directories#dvclock-file) for more info on hash types for specific remote location types.

Just adding this here since I got a bit confused with 2 reviews here: [#1494(comment)](https://github.com/iterative/dvc.org/pull/1494#discussion_r448758146) and [#1494(another comment)](https://github.com/iterative/dvc.org/pull/1494#discussion_r449794597).
Anyway, let me know if there's something wrong in this change and I'll make the changes immediately. :v:

jorgeorpinel commented on content/docs/command-reference/import-url.md:
```diff
@@ -51,8 +51,8 @@ corresponding local path in the <abbr>workspace</abbr>. It records enough
 
-`dvc repro` doesn't check and/or update generated `.dvc` files, use `dvc update`
-on them to bring the import up to date from the external data source.
+`dvc repro` doesn't check or update generated `.dvc` files, use `dvc update` on
```
```suggestion
Note that `dvc repro` doesn't check or update import `.dvc` files, use `dvc update` on
```
Can you please apply this and format the paragraph? A bit out of scope but it relates to something else you added...

jorgeorpinel replied:
Also, in line 48 can you change "In such a `.dvc`" for "In such an import `.dvc`"? Thanks

jorgeorpinel commented on content/docs/command-reference/import-url.md:
```diff
@@ -88,6 +88,12 @@ Specific explanations:
   running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
   URL by appending `/path/to/file` to the `myremote`'s configured base path.
 
+The `deps` field in the generated `.dvc` file contains a hash value along with
+the `path` to track changes in the remote file/directory. The hash type depends
+on the type of remote location (protocol) being tracked. See
+[`dvc.lock` file](/doc/user-guide/dvc-files-and-directories#dvclock-file) for
+info on hash types for specific remote location types.
+
```
Yeah this whole paragraph is probably unnecessary but thanks. The ref. is already too long TBH. Users can find out about .dvc file specs in its dedicated guide, that's not the main purpose for a command ref.

utkarshsingh99 replied:
Okay, so where should I link `dvc.lock file` ?

jorgeorpinel replied:
Why did you decide to edit this command reference if there's no existing places that relate to dvc.lock? I'm going to just merge this with the few typo fixed but I'll have to uncheck that box in the original index so this can be re-evaluated.

utkarshsingh99 replied:
I actually thought that we needed to add some reference to the hash values in each file or describe where it was needed based on the original issue. But does this mean that there is no place we can add a reference to `dvc.lock` or provide info about different types of file hashes in `import-url`?

jorgeorpinel replied:
> does this mean that there is no place we can add a reference to dvc.lock or provide info about different types of file hashes in import-url?

Idk, can you answer that question? 🙂 Let's go back to analyzing the checkboxes of #1469 and please comment there if any one seems unaplicable.

jorgeorpinel commented:
> confused with 2 reviews

Ah yeah, it's the first one: remove whole paragraph. Thanks

jorgeorpinel commented on content/docs/command-reference/import-url.md:
```diff
@@ -88,6 +89,9 @@ Specific explanations:
   running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
   URL by appending `/path/to/file` to the `myremote`'s configured base path.
 
+See [`dvc.lock` file](/doc/user-guide/dvc-files-and-directories#dvclock-file)
+for info on hash types for specific remote location types.
+
```
Again, this ref. is already too long. Let's not add this.

jorgeorpinel replied:
I'll fix it.
```suggestion
```
